### PR TITLE
chore: add cli cmd to incentivize existing packet (async)

### PR DESCRIPTION
**Pull request by colin-axner**
## Description

ref: #871 
closes: #963 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/apps/29-fee/client/cli/cli.go [hunk 1]
```diff
@@ -25,14 +25,14 @@ func GetQueryCmd() *cobra.Command {
 func NewTxCmd() *cobra.Command {
 	txCmd := &cobra.Command{
 		Use:                        "ibc-fee",
-		Short:                      "", // TODO
+		Short:                      "Transaction subcommand for incentivizing relaying IBC packet",
```
Comment (colin-axner) [inline modules/apps/29-fee/client/cli/cli.go:28]:
```suggestion
		Short:                      "Transaction subcommand for incentivizing relaying IBC packets",
```

File: modules/apps/29-fee/client/cli/cli.go [hunk 2]
```diff
@@ -25,14 +25,14 @@ func GetQueryCmd() *cobra.Command {
 func NewTxCmd() *cobra.Command {
 	txCmd := &cobra.Command{
 		Use:                        "ibc-fee",
-		Short:                      "", // TODO
+		Short:                      "Transaction subcommand for incentivizing relaying IBC packets",
```
Comment (AdityaSripal) [inline modules/apps/29-fee/client/cli/cli.go:28]:
```suggestion
		Short:                      "Transaction subcommand for IBC relayer incentivization",
```

File: modules/apps/29-fee/client/cli/tx.go [hunk 3]
```diff
@@ -1,3 +1,99 @@
...
+	flagTimeoutFee = "timeout-fee"
+)
+
+// NewPayPacketFeeAsyncTxCmd returns the command to create a MsgPayPacketFeeAsync
+func NewPayPacketFeeAsyncTxCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "pay-packet-fee [src-port] [src-channel] [sequence]",
+		Short:   "Pay a fee to incentivize an existing IBC packet",
+		Long:    strings.TrimSpace(`Pay a fee to incentivize an existing IBC packet.`),
+		Example: fmt.Sprintf("%s tx pay-packet-fee [src-port] [src-channel] [sequence] --recv-fee 10stake --ack-fee 10stake --timeout-fee 10stake", version.AppName),
```
Comment (AdityaSripal) [inline modules/apps/29-fee/client/cli/tx.go:31]: The example should actually have example packet id rather than parameterizing them correct?

**Comment by seantking** [comment]
Nice one :) 

For this CLI PRs I think it might be useful to add a "how to test locally" section to the PR description. Is this dependent on any other open PRs before it's functional? If so maybe we should leave it open until they're merged. 

Open to ideas, but I think it'd be good to have the CLI PRs mostly functional before merging.

**Review by damiannolan — APPROVED** [review]
Good stuff!

File: modules/apps/29-fee/client/cli/tx.go (hunk 3 above)
Comment (colin-axner) [reply modules/apps/29-fee/client/cli/tx.go:31]: I don't think any of our cli commands do that. It sounds like a good idea to me

**Comment by colin-axner** [comment]
> Is this dependent on any other open PRs before it's functional?

I don't think so. 

> For these CLI PRs I think it might be useful to add a "how to test locally" section to the PR description.

Good idea. Installing the simd and calling this command will allow all the code to execute and msg.ValidateBasic to be called before trying to deliver to a connected node. The actual successful execution of the message is unnecessary imo for this stage of development since that is testing the internals of 29-fee and not the construction of the message via cli

**Comment by seantking** [comment]
> > Is this dependent on any other open PRs before it's functional?
> 
> I don't think so. It worked fine for me
> 
> > For these CLI PRs I think it might be useful to add a "how to test locally" section to the PR description.
> 
> Good idea. Installing the simd and calling this command will allow all the code to execute and msg.ValidateBasic to be called before trying to deliver to a connected node. You should be able to use the example provided by the cli cmd.
> 
> The actual successful execution of the message is unnecessary in my opinion for this stage of development since that is testing the internals of 29-fee and not the construction of the message via cli. I'm just hesitant to delay or spend extra time (I don't have manual testing infra setup to do this quickly) on something that will eventually be tested many times before a final release and that is tested pretty well by our unit tests

That is a good point. I don't wanna slow things down at all. I think the steps you mention here with simd locally are enough for merging this PR. IMO it would be nice to align on the point at which we should try and run this e2e (even locally) before we merge into main (if that is alpha release or otherwise). 

I'll try it out and then approve. I think for these kinds of PR's (user interaction) it would be good for everyone to try locally before approving.

**Review by seantking — APPROVED** [review]

**Pull request merged**